junit-platform-launcher/src/main/java/org/junit/platform/launcher/TagFilter.java

```diff
@@ -117,12 +122,27 @@ public static PostDiscoveryFilter excludeTags(String... tags) throws Preconditio
 	 */
 	public static PostDiscoveryFilter excludeTags(List<String> tags) throws PreconditionViolationException {
 		Preconditions.notEmpty(tags, "tags list must not be null or empty");
-		List<TestTag> activeTags = toTestTags(tags);
-		return descriptor -> FilterResult.includedIf(descriptor.getTags().stream().noneMatch(activeTags::contains));
+		return includeMatching("! (" + orExpressionFor(tags) + ")");
 	}
 
-	private static List<TestTag> toTestTags(List<String> tags) {
-		return tags.stream().map(TestTag::create).collect(toList());
+	/**
+	 * Create an <em>include</em> filter based on the supplied {@code infixTagExpression}.
+	 *
+	 * <p>Containers and tests will only be executed if their tags match the supplied <em>infixTagExpression</em>.
+	 *
+	 * @param infixTagExpression to parse and evaluate against a {@link TestDescriptor}; never {@code null} or empty
+	 * @throws PreconditionViolationException if the supplied infixTagExpression can not be parsed.
+	 * @since 1.1
+	 */
+	public static PostDiscoveryFilter includeMatching(String infixTagExpression) {
```

Comment:
Unless I'm mistaken, the idea was to make `includeMatching` a `private` method -- right?

Reply:
Made it private

Reply:
thanks